app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/HistoryListAdapter.java

```diff
@@ -88,17 +88,35 @@ public void calcTime(Date date, HistoryScanHolder holder) {
 
 
         String secText = String.valueOf(seconds) + " seconds ago";
+
         String minText = String.valueOf(minutes) + " minutes ago";
+        String OneMinText = String.valueOf(minutes) + " minute ago";
+
         String hourText = String.valueOf(hours) + " hours ago";
+        String OneHourText = String.valueOf(hours) + " hour ago";
+
         String dayText = String.valueOf(days) + " days ago";
+        String OneDayText = String.valueOf(days) + " day ago";
```

Comment:
You should use <plural> here to take care of this. This is better for localisation. You should use:
- One
- Two
- Few
- Many
- Other

While English will be easy since two, few, many and other will be the same but this is not the case in every language.